### PR TITLE
[docs] Add a DevelopmentTips.md file.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
We can add more things later. For now, I've added notes from my forum post on using sccache. https://forums.swift.org/t/using-sccache-to-speed-up-compiler-builds/29878

(Review from any 1 or 2 people will do.)

### CODE REVIEW

**Review by keakalamudo — APPROVED** [review]

**Review by shahmishal — APPROVED** [review]

File: docs/DevelopmentTips.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Development Tips
+
+## Compile times
+
+### Be a Ninja
```
Comment (gottesmm) [inline docs/DevelopmentTips.md]: Can you change this to "Build using Ninja"?

File: docs/DevelopmentTips.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Clicking around a little bit, we can find `lib/swift/iphonesimulator/i386/libswiftCore.dylib` as a commonly-depended-upon target. This will perform just what is needed to compile the standard library for i386 and nothing else.
+
+Going further, for various reasons the standard library has lots of warnings. This is actively being addressed, but fixing all of them may require language features, etc. In the mean time, let’s suppress warnings in our build so that we just see the errors. `ninja -nv lib/swift/iphonesimulator/i386/libswiftCore.dylib` will show us the actual commands ninja will issue to build the i386 stdlib. (You’ll notice that an incremental build here is merely 3 commands as opposed to ~150 for `swift-stdlib-iphonesimulator-i386`).
+
+Copy the invocation that has  ` -o <build-path>/swift-macosx-x86_64/stdlib/public/core/iphonesimulator/i386/Swift.o`, so that we can perform the actual call to swiftc ourselves. Tack on `-suppress-warnings` at the end, and now we have the command to just build `Swift.o` for i386 while only displaying the actual errors.
+
+### Use caching
```
Comment (gottesmm) [inline docs/DevelopmentTips.md]: => "Using sccache to cache build artifacts"?

**Review by gottesmm — COMMENTED** [review]
Two nits.

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Review by gottesmm — APPROVED** [review]
LGTM!

**Pull request merged**